Source/CoreGraphicsPolyfill.swift

```diff
@@ -0,0 +1,493 @@
+                deltaAngle = endAngle - startAngle
+                while deltaAngle > 0 { deltaAngle -= 2 * .pi }
+            }
+
+            if abs(deltaAngle) < 1e-6 { return }  // Essentially no arc
+
+            let numSegments = Swift.max(1, Int(ceil(abs(deltaAngle) / (.pi / 2.0))))  // Max 90deg segments
+            let segmentAngleSweep = deltaAngle / CGFloat(numSegments)
+
+            var currentAngle = startAngle
+
+            let initialPoint = CGPoint(
+                x: center.x + radius * cos(currentAngle), y: center.y + radius * sin(currentAngle))
+
+            if path.elements.isEmpty || (path.elements.last?.isCloseSubpath ?? false) {
+                path.move(to: initialPoint)
+            } else if let lastElement = path.elements.last, let lastPoint = lastElement.lastPoint,
+                lastPoint != initialPoint
+            {
+                path.addLine(to: initialPoint)
+            }
+
+            for _ in 0..<numSegments {
+                let nextAngle = currentAngle + segmentAngleSweep
+                let L = radius * (4.0 / 3.0) * tan(abs(segmentAngleSweep) / 4.0)
+
+                let cp1_centerRelative = CGPoint(
+                    x: radius * cos(currentAngle) - L * sin(currentAngle),
+                    y: radius * sin(currentAngle) + L * cos(currentAngle)
+                )
+                let cp2_centerRelative = CGPoint(
+                    x: radius * cos(nextAngle) + L * sin(nextAngle),
+                    y: radius * sin(nextAngle) - L * cos(nextAngle)
+                )
+
+                let endPoint_abs = CGPoint(
+                    x: center.x + radius * cos(nextAngle), y: center.y + radius * sin(nextAngle))
+                let cp1_abs = CGPoint(
+                    x: center.x + cp1_centerRelative.x, y: center.y + cp1_centerRelative.y)
+                let cp2_abs = CGPoint(
+                    x: center.x + cp2_centerRelative.x, y: center.y + cp2_centerRelative.y)
+
+                path.addCurve(to: endPoint_abs, control1: cp1_abs, control2: cp2_abs)
+                currentAngle = nextAngle
+            }
+        }
+    }
+
+    extension CGPath.Element {
+        var lastPoint: CGPoint? {
+            switch self {
+            case .moveToPoint(let p): return p
+            case .addLineToPoint(let p): return p
+            case .addQuadCurveToPoint(_, let p): return p
+            case .addCurveToPoint(_, _, let p): return p
+            case .closeSubpath: return nil
+            }
+        }
+        var isCloseSubpath: Bool {
+            if case .closeSubpath = self { return true }
+            return false
+        }
+    }
+
+#else
+    import CoreGraphics
+    public typealias CGLineJoin = CoreGraphics.CGLineJoin
+    public typealias CGLineCap = CoreGraphics.CGLineCap
+    public typealias CGPathFillRule = CoreGraphics.CGPathFillRule
+    public typealias CGPath = CoreGraphics.CGPath
+    public typealias CGAffineTransform = CoreGraphics.CGAffineTransform
+#endif
```

Comment:
Super nice contriution. Thank you so much @khoi. Did you add some tests to guarantee the implementation is correct?

Reply:
Add tests, add github actions https://github.com/GoodNotes/SVGView/actions/runs/15201364624?pr=2